packages/mobile/src/qrcode/schema.ts

```diff
@@ -29,13 +29,15 @@ export const uriDataFromJson = (obj: object): UriData => {
   }
   return either.right
 }
-export const uriDataFromUrl = (url: string) => uriDataFromJson(parse(url, true).query)
+export const uriDataFromUrl = (url: string) => uriDataFromJson(parse(decodeURI(url), true).query)
 
 enum UriMethod {
   pay = 'pay',
 }
 
 export const urlFromUriData = (data: Partial<UriData>, method: UriMethod = UriMethod.pay) => {
-  const params = new URLSearchParams(Object(data))
-  return `celo://wallet/${method.toString()}?${params.toString()}`
+  const params = new URLSearchParams(
+    Object(Object.entries(data).filter(([k, v], _) => v !== undefined))
```

Comment:
Nit: Can you add a comment I find this difficult to read

Reply:
+1, or consider breaking this apart.